File: SqlParser/SqlParser/ConditionalExpression.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SqlParser
{
    public class ConditionalExpression : Expression
    {
        private string condition;

        public string Condition
        {
            get { return condition; }
            set { condition = value; }
        }


        public Expression Exp
        {
            get
            {
                if (lists.Count > 0)
                {
                    return lists[0];
                }
                else
                {
                    return null;
                }
            }
            set {
                if (lists.Count > 0)
                {
                    lists[0] = value;
                }
                else
                {
                    lists.Add(value);
                }
            }
        }


        public override string Eval(IExpressionContext ctx)
        {
            if (Eval(ctx, condition))
            {
                return Exp.Eval(ctx);
            }
            return null;
        }

        private bool Eval(IExpressionContext ctx, string condition)
        {
            object o = ctx.GetValue(condition);
            if (o == null)
            {
                return false;
            }
            if (o.GetType() == typeof(bool))
            {
                return (bool)o;
            }
            return true;
        }
    }
}

File: app-fw/app-base/src/main/java/org/yy/base/dao/dynamic2/ConditionalExpression.java
package org.yy.base.dao.dynamic2;

public class ConditionalExpression extends Expression {
	private String condition;

	public Expression getExpression() {
		if (lists.size() > 0) {
			return lists.get(0);
		} else {
			return null;
		}
	}

	public String getCondition() {
		return condition;
	}

	public void setCondition(String condition) {
		this.condition = condition;
	}

	public void setExpression(Expression value) {
		if (lists.size() > 0) {
			lists.set(0, value);
		} else {
			lists.add(value);
		}
	}

	public String Eval(ExpressionContext ctx) {
		if (Eval(ctx, condition)) {
			return getExpression().Eval(ctx);
		}
		return null;
	}

	private boolean Eval(ExpressionContext ctx, String condition) {
		Object o = ctx.GetValue(condition);
		if (o == null) {
			return false;
		}
		if (o.getClass() == Boolean.class) {
			return ((Boolean) o).booleanValue();
		}
		return false;
	}
}
